### ISSUE TITLE
alias_method inside refine causes `alias_method': undefined method for module

### ISSUE THREAD

**Issue by deepj** [oracle/truffleruby#1942]
To reproduce:

```ruby
module SnakeCase
  refine ::String do
    def snake_case
    end

    alias_method :snakecase, :snake_case
  end
end
```

Error:

```
ref.rb:6:in `alias_method': undefined method `snake_case' for module `#<Module:0x200>' (NameError)
	from ref.rb:6:in `block in SnakeCase'
	from ref.rb:2:in `refine'
	from ref.rb:2:in `SnakeCase'
	from ref.rb:1:in `<main>'
```

**Comment by alecgorge** [comment]
I am experimenting with TruffleRuby as well and ran into the same issue. `alias`/`alias_method` works for existing methods but not methods declared inside the refinement.

Here is a simple set of cases for reproduction:

```ruby
module ExampleRefinements
  # DOESN'T WORK
  refine Integer do
    def testing
      self.abs
    end
    # or alias_method
    alias test testing
  end

  # WORKS
  # refine Integer do
  #   def testing
  #     self.abs
  #   end
  #
  #   def test
  #     testing
  #   end
  # end

  # WORKS
  # refine Integer do
  #   def testing
  #     self.abs
  #   end
  #   # or alias_method
  #   alias test abs
  # end
end

using ExampleRefinements
puts -1.testing
puts -1.test
puts -1.testing == -1.test
```

Expected output:

```
1
1
true
```

**Comment by deepj** [comment]
@eregon this is fixed in https://github.com/oracle/truffleruby/pull/1988

**Comment by eregon** [comment]
(this was actually fixed before, in d31d48fe4f)
